### PR TITLE
hardshrink and Hardshrink: add class, threshold default 0.5

### PR DESCRIPTION
**Pull request by zhupengyang**
### PR types

Others

### PR changes

APIs

### Describe

- add Hardshrink class
- threshold default is 0.5

### CODE REVIEW

**Review by hong19860320 — DISMISSED** [review]
LGTM

File: python/paddle/nn/layer/activation.py [hunk 1]
```diff
@@ -30,6 +31,53 @@
...
+
+        .. code-block:: python
+
+        import paddle
+        import numpy as np
+
+        paddle.disable_static()
+
+        x = paddle.to_variable(np.array([-1, 0.3, 2.5]))
+        hd = paddle.nn.Hardshrink()
```
Comment (T8T9) [inline python/paddle/nn/layer/activation.py:68]: 这个是不是用 m = paddle.nn.Hardshrink(), torch 中统一都是m

Comment (zhupengyang) [reply python/paddle/nn/layer/activation.py]: Done

**Comment by T8T9** [comment]
LGTM

**Review by hong19860320 — APPROVED** [review]
LGTM

**Pull request merged**